sys-block/open-iscsi: fix build against musl

stintel opened the pull request:
Add an upstream patch that fixes build on musl.

thesamesam commented:
Thanks! I added a `Closes:` tag for https://bugs.gentoo.org/908587 and in a followup commit, I added some links to the patch too.

stintel commented:
> Thanks! I added a `Closes:` tag for https://bugs.gentoo.org/908587 and in a followup commit, I added some links to the patch too.

Seems I totally missed that bug. Sorry about that. Will try to look harder next time!
